server/src/test/java/org/opensearch/search/pipeline/SearchPipelineServiceTests.java

```diff
@@ -1079,18 +1174,140 @@ public void testStats() throws Exception {
         }
     }
 
+    public void testStatsEnabledIgnoreFailure() throws Exception {
+        SearchRequestProcessor throwingRequestProcessor = new FakeRequestProcessor("throwing_request", "1", null, false, r -> {
+            throw new RuntimeException();
+        });
+        Map<String, Processor.Factory<SearchRequestProcessor>> requestProcessorsEnableIgnoreFailure = Map.of(
+            "successful_request",
+            (pf, t, i, f, c, ctx) -> new FakeRequestProcessor("successful_request", "2", null, true, r -> {}),
+            "throwing_request",
+            (pf, t, i, f, c, ctx) -> throwingRequestProcessor
+        );
+        SearchResponseProcessor throwingResponseProcessor = new FakeResponseProcessor("throwing_response", "3", null, false, r -> {
+            throw new RuntimeException();
+        });
+        Map<String, Processor.Factory<SearchResponseProcessor>> responseProcessorsEnableIgnoreFailure = Map.of(
+            "successful_response",
+            (pf, t, i, f, c, ctx) -> new FakeResponseProcessor("successful_response", "4", null, true, r -> {}),
+            "throwing_response",
+            (pf, t, i, f, c, ctx) -> throwingResponseProcessor
+        );
```

Comment:
I don't understand what's going on in this test. It looks like your successful processors are ignoring failures (but not failing), while your failing processors are not ignoring failure. 

Functionally, I don't see a difference in execution between this and `testStats`.

Reply:
My intention is to test the stats when the ignoreFailure is enabled. I will turn on the ignoreFailure for the throwingProcessor as well.